pkg/generate/generate.go

```diff
@@ -107,6 +109,28 @@ func (c *Controller) applyGenerate(resource unstructured.Unstructured, gr kyvern
 		return nil, fmt.Errorf("policy %s, dont not apply to resource %v", gr.Spec.Policy, gr.Spec.Resource)
 	}
 
+	for i, r := range engineResponse.PolicyResponse.Rules {
+		if !r.Success {
+			grList, err := c.kyvernoClient.KyvernoV1().GenerateRequests(config.KubePolicyNamespace).List(metav1.ListOptions{})
+			if err != nil {
+				continue
+			}
+			for _, v := range grList.Items {
+				if engineResponse.PolicyResponse.Policy == v.Spec.Policy && engineResponse.PolicyResponse.Resource.Name == v.Spec.Resource.Name && engineResponse.PolicyResponse.Resource.Kind == v.Spec.Resource.Kind && engineResponse.PolicyResponse.Resource.Namespace == v.Spec.Resource.Namespace{
+					err :=c.kyvernoClient.KyvernoV1().GenerateRequests(config.KubePolicyNamespace).Delete(v.GetName(),&metav1.DeleteOptions{})
+					if err != nil {
+						logger.Error(err, "failed to update gr")
+					}
+				}
+			}
+			if len(engineResponse.PolicyResponse.Rules) > 1 {
+				engineResponse.PolicyResponse.Rules = append(engineResponse.PolicyResponse.Rules[:i], engineResponse.PolicyResponse.Rules[i+1:]...)
+				continue
+			}else if len(engineResponse.PolicyResponse.Rules) == 1 {
+				engineResponse.PolicyResponse.Rules = []response.RuleResponse{}
```

Comment:
What is this `if..else` does? Why does it truncate rules?

Reply:
if a rule is false then we will turn cat it from the engine reponse